loopflow-computation/src/main/java/com/farao_community/farao/loopflow_computation/LoopFlowComputation.java

```diff
@@ -6,179 +6,102 @@
 
-    public Map<String, Double> calculateLoopFlowsApproximation(Network network) {
-        Map<Cnec, Double> frefResults = computeRefFlowOnCurrentNetwork(network); //get reference flow
-        Map<Cnec, Double> loopFlowShifts = buildLoopflowShiftsApproximation(crac);
-        return buildLoopFlowsFromReferenceFlowAndLoopflowShifts(frefResults, loopFlowShifts); //compute loopflow
-    }
+        SystematicSensitivityInterface systematicSensitivityInterface = SystematicSensitivityInterface.builder()
+            .withDefaultParameters(sensitivityComputationParameters)
+            .withPtdfSensitivities(glskProvider, cnecs)
+            .build();
 
-    public Map<Cnec, Map<Country, Double>> computePtdfOnCurrentNetwork(Network network) {
-        Map<Cnec, Map<Country, Double>> ptdfs = new HashMap<>();
-        Set<Cnec> preventivecnecs = crac.getCnecs(crac.getPreventiveState());
-        SensitivityFactorsProvider factorsProvider = net -> generateSensitivityFactorsProvider(net, preventivecnecs, glskProvider);
-        SensitivityComputationResults sensiResults = SensitivityComputationService.runSensitivity(network, network.getVariantManager().getWorkingVariantId(), factorsProvider);
-        sensiResults.getSensitivityValues().forEach(sensitivityValue -> addSensitivityValue(sensitivityValue, crac, ptdfs));
-        return ptdfs;
-    }
+        SystematicSensitivityResult ptdfsAndRefFlows = systematicSensitivityInterface.run(network, Unit.MEGAWATT);
 
-    private List<SensitivityFactor> generateSensitivityFactorsProvider(Network network, Set<Cnec> cnecs, GlskProvider glskProvider) {
-        List<SensitivityFactor> factors = new ArrayList<>();
-        Map<String, LinearGlsk> mapCountryLinearGlsk = glskProvider.getAllGlsk(network);
-        cnecs.forEach(cnec -> mapCountryLinearGlsk.values().stream()
-                .map(linearGlsk -> new BranchFlowPerLinearGlsk(new BranchFlow(cnec.getId(), cnec.getName(), cnec.getNetworkElement().getId()), linearGlsk))
-                .forEach(factors::add));
-        return factors;
+        return buildLoopFlowsFromReferenceFlowAndPtdf(ptdfsAndRefFlows, network);
     }
 
-    private void addSensitivityValue(SensitivityValue sensitivityValue, Crac crac, Map<Cnec, Map<Country, Double>> ptdfs) {
-        String cnecId = sensitivityValue.getFactor().getFunction().getId();
-        Cnec cnec = crac.getCnec(cnecId);
-        String glskId = sensitivityValue.getFactor().getVariable().getId();
-        double ptdfValue = sensitivityValue.getValue();
-        if (!ptdfs.containsKey(cnec)) {
-            ptdfs.put(cnec, new HashMap<>());
-        }
-        ptdfs.get(cnec).put(glskIdToCountry(glskId), ptdfValue);
-    }
+    public LoopFlowResult buildLoopFlowsFromReferenceFlowAndPtdf(SystematicSensitivityResult alreadyCalculatedPtdfAndFlows, Network network) {
+        return buildLoopFlowsFromReferenceFlowAndPtdf(alreadyCalculatedPtdfAndFlows, alreadyCalculatedPtdfAndFlows, network);
```

Comment:
Remove this duplication of parameters

Reply:
The initial idea was to use this method when loop-flows are calculating using the approximation where PTDFs are not recomputed, in that case we might have wanted the PTDFs and the ref flows to come from two differents objects.

However, due to the size of the SystematicSensitivityResult, we prefer not to store two of those object, so instead of using this method, we compute the loopflows differently.

I remove this method then